internal/controller/pipelinerollout_controller.go

```diff
@@ -492,6 +493,18 @@ func (r *PipelineRolloutReconciler) processExistingPipeline(ctx context.Context,
 		}
 	}
 
+	if inProgressStrategy != apiv1.UpgradeStrategyNoOp {
```

Comment:
```suggestion
	// don't risk out-of-date cache while performing PPND or Progressive strategy - get 
	// the most current version of the Pipeline just in case
	if inProgressStrategy != apiv1.UpgradeStrategyNoOp {
```